### DISCUSSION TITLE
Exogenous variable use on GluonTS

### DISCUSSION THREAD

**Discussion by spark49** [awslabs/gluonts#1589] [Q&A]
Hi everyone, i was trying to use GluonTS to train and predicted different timeseries by using an exogenous variable.

My dataset consist of 100 time_series, each of which has 60 data_point. 

I would like to predict 20 point and so use 40 point of each series to train the model.

I had splitted the dataset as following:
```python 
print(target_train.shape)
print(exogenous_static_real_train.shape)
print(target_test.shape)
print(exogenous_static_real_test.shape)

(100, 40)
(100, 40)
(100, 60)
(100, 60)
```

Defined the training dataset:
```python
train_ds = ListDataset([
    {
        FieldName.TARGET: x,
        FieldName.START: start_train, #first date of each series
        FieldName.FEAT_STATIC_REAL: fsr
    }
   for x, fsr in zip(target_train, exogenous_static_real_train)
], freq = "M")
```

Defined and trained the estimator:
```python
estimator = deepar.DeepAREstimator(
            freq='M',
            prediction_length=20,
            use_feat_static_real=True,
            trainer=Trainer(epochs=10),
        )
predictor = estimator.train(training_data=train_ds)
```

Defined the test dataset in this way:

```python
test_ds = ListDataset([
    {
        FieldName.TARGET: x,
        FieldName.START: start_test, #41th date (equal for each series)
        FieldName.FEAT_STATIC_REAL: fsr
    }
   for x, fsr in zip(target_test, exogenous_static_real_test)
], freq = "M")
```

However, when i go to perform the evaluation (I would like to consider also the future value of exogenous variable):
```python
from gluonts.evaluation import make_evaluation_predictions
forecast_it, ts_it = make_evaluation_predictions(
    dataset=test_ds,  # test dataset
    predictor=predictor,  # predictor
    num_samples=100,  # number of sample paths we want for evaluation
)
forecast_entry, ts_entry = next(forecast_it), next(ts_it)
```

The following error is raised:
```python
MXNetError: MXNetError: Shape inconsistent, Provided = [160,60], inferred shape=(160,80)

```

Could someome help me on this? I would like to understand how can i configure it with exogenous variable, since if i remove it from the following test i did not get any error. It seems that it is not able to accept any of the future value.

Also, how can i apply them to prediction? Is GluonTS restricted to use always the train interval and is it not possible to provide know future exogenous variable?

Furthermore, if i change the evaluation dataset in:
```python
test_ds = ListDataset([
    {
        FieldName.TARGET: x,
        FieldName.START: start_test, #41th date (equal for each series)
        FieldName.FEAT_STATIC_REAL: fsr
    }
   for x, fsr in zip(target_test, **exogenous_static_real_train**)
], freq = "M")
```

Everything works fine.

Thank you very much.

**Comment by lostella** [comment]
Could you try setting the "start" field in `test_ds` equal to the one for `train_ds`? If I understand correctly how you got the two datasets, I think they start at the same timestamp.
